tests/handlers/test_receipts.py

```diff
@@ -26,24 +26,25 @@ def prepare(self, reactor, clock, hs):
         self.event_source = hs.get_event_sources().sources.receipt
 
     def test_filters_out_private_receipt(self):
-        self._test_filters_private(
-            [
-                {
-                    "content": {
-                        "$1435641916114394fHBLK:matrix.org": {
-                            ReceiptTypes.READ_PRIVATE: {
-                                "@rikj:jki.re": {
-                                    "ts": 1436451550453,
-                                }
+        events = [
+            {
+                "content": {
+                    "$1435641916114394fHBLK:matrix.org": {
+                        ReceiptTypes.READ_PRIVATE: {
+                            "@rikj:jki.re": {
+                                "ts": 1436451550453,
                             }
                         }
-                    },
-                    "room_id": "!jEsUZKDJdhlrceRyVU:example.org",
-                    "type": "m.receipt",
-                }
-            ],
-            [],
-        )
+                    }
+                },
+                "room_id": "!jEsUZKDJdhlrceRyVU:example.org",
+                "type": "m.receipt",
+            }
+        ]
+        original_events = deepcopy(events)
+        self._test_filters_private(events, [])
+        # Since the events are fed in from a cache they should not be modified.
+        self.assertEqual(events, original_events)
```

Comment:
I think it would be a bit nicer to have a separate test case for mutating rather than using an existing one